go-controller/pkg/node/gateway_iptables.go

```diff
@@ -467,6 +467,40 @@ func getLocalGatewayNATRules(ifname string, cidr *net.IPNet) []nodeipt.Rule {
 			Protocol: protocol,
 		},
 	}
+	// the following rules are actively used only for the UDN Feature:
+	// -A POSTROUTING -s 169.254.0.0/29 -j RETURN ---> UDN
+	// -A POSTROUTING -s 169.254.0.0/16 -j MASQUERADE ---> UDN
+	// NOTE: Ordering is important here, the RETURN must come before
+	// the MASQUERADE rule. Please don't change the ordering.
+	srcUDNMasqueradePrefix := config.Gateway.V4MasqueradeSubnet
+	// defaultNetworkReservedMasqueradePrefix contains the first 6IPs in the masquerade
+	// range that shouldn't be MASQUERADED. Hence /29 and /125 is intentionally hardcoded here
+	defaultNetworkReservedMasqueradePrefix := config.Gateway.MasqueradeIPs.V4HostMasqueradeIP.String() + "/29"
+	if protocol == iptables.ProtocolIPv6 {
+		srcUDNMasqueradePrefix = config.Gateway.V6MasqueradeSubnet
+		defaultNetworkReservedMasqueradePrefix = config.Gateway.MasqueradeIPs.V6HostMasqueradeIP.String() + "/125"
+	}
+	rules = append(rules,
+		nodeipt.Rule{
+			Table: "nat",
+			Chain: "POSTROUTING",
+			Args: []string{
+				"-s", defaultNetworkReservedMasqueradePrefix,
+				"-j", "RETURN",
+			},
+			Protocol: protocol,
+		},
+		nodeipt.Rule{
```

Comment:
I thought you were only going to add the masquerade rule if network segmentation was on. If you dont want to split them up then you can just put it back with the flag and we can fix it later.

Looking at the POSTROUTING chain:
```
Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination         
OVN-KUBE-EGRESS-IP-MULTI-NIC  all  --  anywhere             anywhere            
OVN-KUBE-EGRESS-SVC  all  --  anywhere             anywhere            
OVN-KUBE-SNAT-MGMTPORT  all  --  anywhere             anywhere            
DOCKER_POSTROUTING  all  --  anywhere             fedora         
```
It would be probably nicer to put these in OVN_KUBE-SNAT-MASQ or something.

Reply:
Ah ouch I missed the mark.. I didn't realise the RETURN rule is what you wanted always on by default.. 

Sorry yea let me fix this up and also can add a new chain for MASQUERADE-UDN ? Martin also asked that so probably best to add that now

Reply:
going with `OVN-KUBE-UDN-SNAT-MASQ` from POSTROUTING and then gonna add both rules there under the feature flag. I will open a card to track "disabling the feature should cleanup all UDN stuff".